pkg/reconciler/taskrun/resources/taskref_test.go

```diff
@@ -1016,6 +1039,288 @@ func TestGetTaskFunc_VerifyError(t *testing.T) {
 	}
 }
 
+func TestGetTaskFunc_V1Task_VerifyNoError(t *testing.T) {
```

Comment:
The tests in this PR are all copied from current v1beta1 tasks. I tried to merge them into 1 test but seems not so straightforward. Maybe the easiest way for now is to have separate tests

Reply:
I'm guessing this tests might change a bit once we swap the storage version to v1 - I think it's worth checking them to see if we can merge the tests or at least come up with a few helper methods to reduce the amount of copied code between tests

Reply:
Yes, I will pair with @JeromeJu on fixing the tests for the swap PR. 
I tried to merge the v1 tests into v1beta1 tests but it still requires lots of test code changes. But helper functions may help.

Reply:
This could be tackled as part of https://github.com/tektoncd/pipeline/issues/5820